### PR TITLE
sample: nrf9160: lwm2m_client: Add shell cmd to turn off modem

### PR DESCRIPTION
**Pull request by eibe85**
Carriers are not happy with us for not shutting down modem properly
during testing. This code will allow us to detach from network cleanly
during testing. Strictly speaking something like this would be nice to
add as a library function that could be added to samples with a Kconfig.

Signed-off-by: Eivind Berntsen <eivind.berntse@nordicsemi.no>

### CODE REVIEW

File: samples/nrf9160/lwm2m_client/src/shell_cmds/shell_cmds.c [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      printk("The modem powered off\n");
+    }
+    else {
+        printk("\n***** WARNING! Modem did not shutdown correctly. error: %d *****\n", err);
+    }
+
+    return 0;
+}
+
+SHELL_CMD_REGISTER(shutdown_modem, NULL, "Turn off modem to do a clean detaach from carreir network", cmd_shutdown_modem);
```
[the remark this answers is not in the document]
Comment (VeijoPesonen) [reply samples/nrf9160/lwm2m_client/src/shell_cmds/shell_cmds.c:25]: I don't know if this plays perfectly with the client but I assume solving a real issue is the thing here.

**Review by VeijoPesonen — CHANGES_REQUESTED** [review]

File: samples/nrf9160/lwm2m_client/src/shell_cmds/shell_cmds.c [hunk 2]
```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
```
Comment (rlubos) [inline samples/nrf9160/lwm2m_client/src/shell_cmds/shell_cmds.c:2]: 2022

File: samples/nrf9160/lwm2m_client/src/shell_cmds/shell_cmds.c [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+
+#include <shell/shell.h>
+#include <modem/lte_lc.h>
+
+int cmd_shutdown_modem(const struct shell *shell, size_t argc, char **argv)
+{
+    int err = 0;
```
Comment (rlubos) [inline samples/nrf9160/lwm2m_client/src/shell_cmds/shell_cmds.c:13]: You should use tab for indentation.

File: samples/nrf9160/lwm2m_client/src/shell_cmds/shell_cmds.c [hunk 4]
```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
```
Comment (rlubos) [inline samples/nrf9160/lwm2m_client/src/shell_cmds/shell_cmds.c:6]: One empty line is enough.

File: samples/nrf9160/lwm2m_client/src/shell_cmds/shell_cmds.c [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+#include <shell/shell.h>
+#include <modem/lte_lc.h>
+
+int cmd_shutdown_modem(const struct shell *shell, size_t argc, char **argv)
+{
+    int err = 0;
+    err = lte_lc_power_off();
+    if (err == 0) {
+      printk("The modem powered off\n");
```
Comment (rlubos) [inline samples/nrf9160/lwm2m_client/src/shell_cmds/shell_cmds.c:16]: You should rather use `shell_print()` as this is the response for the command.

File: samples/nrf9160/lwm2m_client/src/shell_cmds/shell_cmds.c [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    }
+    else {
+        printk("\n***** WARNING! Modem did not shutdown correctly. error: %d *****\n", err);
```
Comment (rlubos) [inline samples/nrf9160/lwm2m_client/src/shell_cmds/shell_cmds.c:19]: ditto

File: samples/nrf9160/lwm2m_client/src/shell_cmds/shell_cmds.c (hunk 1 above)
Comment (rlubos) [inline samples/nrf9160/lwm2m_client/src/shell_cmds/shell_cmds.c:25]: nit, but perhaps `modem_shutdown` would be a better command - easier to find alphabetically.

**Review by rlubos — COMMENTED** [review]
I agree it'd be best to have some generic shell module to deal with the modem, but as a temporary solution for the sample this seems ok (unless we start duplicating the solution across the samples).

**Comment by SeppoTakalo** [comment]
This should not be LwM2M specific.
I proposed similar in https://github.com/nrfconnect/sdk-nrf/pull/7054 so that any application that uses shell is benefiting those.

**Comment by VeijoPesonen** [comment]
Replaced with PR #7054

**Pull request closed without merging**